### ISSUE TITLE
Improvements in the UI

### ISSUE THREAD

**Issue by VibhorCodecianGupta** [coding-blocks/codingblocks.online.projectx#190]
I discovered some potential discrepancies and scopes of improvement in the UI throughout the project. 
**Some of these are just suggestions, and will be resolved iff the maintainers agree to those too.**

Issues 4-8 are grammar and vocabulary related issues.

_**1.**_
![1](https://user-images.githubusercontent.com/29275810/42327643-d7e98b8c-8089-11e8-9743-b73f064c898f.png)
Not sure why this is happening, the same rating of 5 is displayed as 5 somewhere and 5.0 somewhere else

_**2.**_
![2](https://user-images.githubusercontent.com/29275810/42327690-f5014f7a-8089-11e8-8f94-458982bdc50f.png)
**Suggestion**
The stars show the average user rating, the metric after the start also shows the average rating and the user rating is just shown in the line above. It's pretty confusing at a first glance. A possible way could be to display another line of star rating below the average stars and mark them as the user rating. Those stars would display what the user just rated. (IMDb does this.)

_**3.**_
![19](https://user-images.githubusercontent.com/29275810/42327723-0cbe5e50-808a-11e8-932a-2e2ffbea89d2.png)
The number of users who have reviewed is displaying a little too close to the stars, a line break would be neater (tried it, it does look better)

_**4.**_
![3](https://user-images.githubusercontent.com/29275810/42327949-b442c094-808a-11e8-9505-d90d79a77766.png)
Revisit and Revise:
remove the 'then' after experienced developer (grammar issue).
Evaluation and Certification:
It should say 'in your path **to** becoming' and something seems missing after Coding Blocks in the same sentence. Also, accredit is a more apt term than verify.

_**5.**_
![4](https://user-images.githubusercontent.com/29275810/42328090-12eb0156-808b-11e8-8131-8e77fde60eee.png)
Seems like there is no need for 'which is' here. And it should be 'hundreds' instead of 100s, it looks odd to use alphanumerics in text like that.

_**6.**_
![5](https://user-images.githubusercontent.com/29275810/42328206-516f7524-808b-11e8-99ed-732bf91220b4.png)
Comprehensive Learning:
It should be 'help you keep your skills sharp' instead of 'help you to'
24*7 Mentor Support:
It should be '24x7' instead of the '*'. Also, there is the use of the word 'online' twice here which seems like an overkill and bad sentence structure. I framed it like this -
`Our online programming courses are a flexible alternative that provides access to our in-depth curriculum-complete with real-time support from instructors, our own industry-tested learning platform, hands-on assignments and much more, all from the comfort of your abode.`

_**7.**_
![6](https://user-images.githubusercontent.com/29275810/42328400-c69dc832-808b-11e8-9e3d-2b0f64fc98b5.png)
This seems very trivial, but generally, websites always have it. The copyright icon just after the word Copyright and before the year.

_**8.**_
![7](https://user-images.githubusercontent.com/29275810/42328462-ed04566c-808b-11e8-8183-485023b2d867.png)
Alright so this is an issue with both the detailed summary and the small summaries displayed on course cards of both the online and offline Java courses, it should be 24x7 instead of 24*7.

_**9.**_
![8](https://user-images.githubusercontent.com/29275810/42328528-1beba462-808c-11e8-8f6a-34a896d113da.png)
Both All courses and Recommended courses components have this issue, the ratings displayed on some of the cards wrap to the next line.

_**10.**_
![9](https://user-images.githubusercontent.com/29275810/42328796-cfbd3960-808c-11e8-9e13-50981f4e4748.png)
![10](https://user-images.githubusercontent.com/29275810/42328893-11975fe6-808d-11e8-954c-6f44892b8299.png)
**Suggestion**
It's a qualitative input, the word 'Content' strikes a bit out of place. The word 'module' could be used instead to say '2 of 10 modules completed'. Similarly for the Table of Content. Either it should be 'Course Contents' (like coursera does it) or 'Course modules'.

_**11.**_
![11](https://user-images.githubusercontent.com/29275810/42328990-5d1343a4-808d-11e8-946d-325ad825d9bb.png)
Here the icon used for drop-down could be a down-pointing arrow (font awesome) and an up-pointing arrow when the menu is dropped down. Also, right now the icon is a fixed distance from where the text ends. So the `>` icon moves more to the right if Python 2.7 is selected in place of C++ because its a longer word. The icon should be fixed on the select box.

_**12.**_
![12](https://user-images.githubusercontent.com/29275810/42329677-e788bbee-808e-11e8-9989-82d50b3526d8.png)
It should say `Reply` when there is just one. 
**Suggestion**
Also, to make it intuitive, the 'x Replies' should also redirect to the blog page (something like quora). If not, it could probably be made in a smaller font to indicate intuitively that it's not a link.

_**13.**_
![13](https://user-images.githubusercontent.com/29275810/42329704-fb6fc2ce-808e-11e8-82ea-0e7c4aa928e4.png)
Not sure if this an issue for this particular project, but displaying the metadata of the image file wouldn't be desirable.

_**14.**_
![15](https://user-images.githubusercontent.com/29275810/42329761-1db475aa-808f-11e8-96b9-51fb2e1e6077.png)

**Suggestion**
The tab is named 'Solve your Doubts' and the heading just below is 'Ask your Doubts' which seems odd because they are one and the same thing in repetition. The tab heading could either be 'Ask Your Doubts' (which looks far from precise for a tab heading) or 'Forums' (like coursera does it).
Below is an image showing how it'll look. Note that this change will be made only after discussion.

![14](https://user-images.githubusercontent.com/29275810/42330045-de0ab0d0-808f-11e8-98a7-cb44af8bd683.png)

_**15.**_
![16](https://user-images.githubusercontent.com/29275810/42330015-ca2a17cc-808f-11e8-9270-0e728363b429.png)
The headings here should be centre aligned in there respective cols.

_**16.**_
![20](https://user-images.githubusercontent.com/29275810/42330100-fbadf318-808f-11e8-830c-bd01ee52616b.png)
This should be `1x` instead of `1X`. Also, if this video right here is the standard of the self hosted videos, then the playback speed icon should be moved elsewhere (probably top right) to avoid clashing with the volume/full screen icons. Again, it depends on what the standard of the icon placing would be in the video.

_**17.**_
![9](https://user-images.githubusercontent.com/29275810/42378854-6ab85ef4-8146-11e8-8c0e-b1bf8fbab774.png)
The `%` sign wraps to the next line. This is undesirable.

@abhishek97 @championswimmer

**Comment by witty123** [comment]
Thanks for the comprehensive list @VibhorCodecianGupta 
1st issue has already been fixed.
2nd issue is something that you can work on, if the UX improves then we'll merge the PR
3rd - 9th, you can work on these issues
10th seems a little confusing, will need more clarification
11th - 13th, agreed
14th, please put this on hold, need some more time to think about it
15th, agreed
16th, approved in the PR, will need screenshots for mobile screen as well
17th, agreed

**Comment by championswimmer** [comment]
Awarding you a 300 point bounty for finding so many issues !!

**Comment by VibhorCodecianGupta** [comment]
@witty123 
10th issue simply says that the word 'Content' used in `2 of 10 Content completed` sounds a bit odd. Basically, the word can be replaced by a better word altogether, something which is self-explanatory. That's why I thought the word 'module' is appropriate. It is very widely used when we talk about breaking up a course into parts, which is also something being done in the course contents being broken into 10 parts or 'modules' if you will. Hence the word 'module' seems more convincing.

**Comment by VibhorCodecianGupta** [comment]
@witty123 made some updates in #193, please review :)

**Comment by witty123** [comment]
So suppose you've 10 different sections inside a course. Now each section can have multiple videos/lectures/docs/problems, what would you name them ?
Modules -> section
section content -> ?

**Comment by witty123** [comment]
for 14th issue, you can change the title to "Ask your Doubts"

**Comment by VibhorCodecianGupta** [comment]
@witty123 
It could be something like Course -> modules -> tasks.
So a course has modules, and each module 'tasks' to be completed be it videos/lectures/docs/problems

**Comment by VibhorCodecianGupta** [comment]
@witty123 @championswimmer 
So I messed up a little. I was supposed to claim 300 BOUNTY for reporting this issue and 500 BOUNTY for the PR #193 . Instead, I ended up claiming 300 BOUNTY for #193 . It would be great if you could reject that bounty or edit it to be 500, so I can make the 300 claim on the right issue 😅
